pinot-core/src/main/java/org/apache/pinot/core/segment/processing/mapper/SegmentMapper.java

```diff
@@ -105,8 +108,12 @@ public SegmentMapper(List<RecordReader> recordReaders, SegmentProcessorConfig pr
    */
   public Map<String, GenericRowFileManager> map()
       throws Exception {
+    Consumer<Object> observer = _processorConfig.getProgressObserver();
+    int totalCount = _recordReaders.size();
+    int count = 1;
     GenericRow reuse = new GenericRow();
     for (RecordReader recordReader : _recordReaders) {
+      observer.accept(String.format("Doing map phase on data from RecordReader: %d/%d", count++, totalCount));
```

Comment:
It seems to me that it's hard to get the meaning of those two numbers. Maybe change it to `RecordReader %d (%d in total)` or something similar?

Reply:
sounds good!